checker/match.go

```diff
@@ -174,3 +177,47 @@ func matchHTTPMethods(methods []string, reqMethod string) bool {
 	log.Debug("HTTP Method not matched.")
 	return false
 }
+
+func matchL4Protocol(rule *proto.Rule, dest *authz.AttributeContext_Peer) bool {
+	// Extract L4 protocol type of socket address for destination peer context. Match against rules.
+	if dest == nil {
+		log.Warn("nil request destination peer.")
+		return false
+	}
+
+	// default protocol is TCP
+	reqProtocol := "TCP"
+	if a := dest.GetAddress(); a != nil {
+		if sa := a.GetSocketAddress(); sa != nil {
+			reqProtocol = sa.GetProtocol().String()
+		}
+	}
```

Comment:
You can just do
```
reqProtocol := dest.GetAddress().GetSocketAddress().GetProtocol().String()
```

If you take a peek at these `Get` functions on Protobuf, they include the check for `nil`.

Also note that protobuf encoding will not include any "zero" values on the wire. Since TCP is the zero value for Protocol, the above snippet will return "TCP" if anything in the chain is `nil`, so no need to initialize a default.

Reply:
Cool. Good to know that. Fixed.